src/experimental/tools/videoThumbnailLoader/remove_buffer.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  combineLatest as observableCombineLatest,
+  Observable,
+  of as observableOf,
+} from "rxjs";
+import { QueuedSourceBuffer } from "../../../core/source_buffers";
+
+/**
+ * Remove buffer around wanted time
+ * @param {HTMLMediaElement} videoElement
+ * @param {Object} sourceBuffer
+ * @param {Number} time
+ * @returns {Observable}
+ */
+export default function removeBufferForTime$(videoElement: HTMLMediaElement,
+                                             sourceBuffer: QueuedSourceBuffer<Uint8Array>,
+                                             time: number): Observable<unknown> {
+  return (videoElement.buffered.length > 0) ?
+    observableCombineLatest([
+      ((time - 60 * 10) > 0) ?
+        sourceBuffer.removeBuffer(0, time - (60 * 10)) :
+        observableOf(null),
+      ((time + 60 * 10) < videoElement.duration) ?
+        sourceBuffer.removeBuffer(time + (60 * 10), videoElement.duration) :
+        observableOf(null)]) :
+    observableOf(null);
+}
```

Comment:
I don't get what this function does from its code either (60 * 10?)

Reply:
Is it good now ?

Reply:
Ok ok, I understand it.
I still have some reservation about the relation between the name of the function and its arguments (`removeBuffer(videoElement, sourceBuffer, 10)` -> will remove the buffer from 10 minutes 10, this does not seem logical, I would have guessed that it removes the buffer at 10 seconds).

Reply:
Still the case, the function signature is not really clear about what it does.